### ISSUE TITLE
[JENKINS-7004] cause not working in parameterized projects

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-23#8359]
<p>The documentation in <em>trigger builds remotely</em> says</p>

<blockquote><p>Optionally append <tt>&cause=Cause+Text</tt> to provide text that will be included in the recorded build cause.</p></blockquote>

<p>The <tt>cause</tt> doesn't work for parameterized builds.  </p>

<p>Also the documentation should state that the trigger URL in that case has <tt>buildWithParameters</tt> not <tt>build</tt>...</p>

<p>Already requested in <a href="http://wiki.jenkins-ci.org/display/JENKINS/Parameterized+Build?focusedCommentId=40042520#comment-40042520" class="external-link" target="_blank" rel="nofollow noopener">wiki</a></p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=anshnd">anshnd</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-7004" target="_blank">cause not working in parameterized projects</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: core
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2010-07-18T08:59:03+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 3
<li><b>imported</b>: 2025-11-23
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>The documentation in <em>trigger builds remotely</em> says</p>

<blockquote><p>Optionally append <tt>&amp;cause=Cause+Text</tt> to provide text that will be included in the recorded build cause.</p></blockquote>

<p>The <tt>cause</tt> doesn't work for parameterized builds.  </p>

<p>Also the documentation should state that the trigger URL in that case has <tt>buildWithParameters</tt> not <tt>build</tt>...</p>

<p>Already requested in <a href="http://wiki.jenkins-ci.org/display/JENKINS/Parameterized+Build?focusedCommentId=40042520#comment-40042520" class="external-link" target="_blank" rel="nofollow noopener">wiki</a></p></pre>
</details>
</details>
<details><summary><i>1 attachment</i></summary>

- [jenkinstestharness.txt](https://issues.jenkins.io/secure/attachment/46342/jenkinstestharness.txt)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>anshnd:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-7004?focusedId=140232&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-140232">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>assigning to parameters and not to unrelated parameterized-trigger</p>
</pre>
</details>

</ul>
</details>

<p>assigning to parameters and not to unrelated parameterized-trigger</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-7004?focusedId=140245&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-140245">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in hudson
User: : mindless
Path:
 trunk/hudson/main/core/src/main/java/hudson/model/AbstractProject.java
 trunk/hudson/main/core/src/main/java/hudson/model/ParametersDefinitionProperty.java
 trunk/hudson/main/core/src/main/resources/hudson/model/BuildAuthorizationToken/config.jelly
 trunk/www/changelog.html
<a href="http://jenkins-ci.org/commit/33032" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/33032</a>
Log:
<span class="error">&#91;FIXED JENKINS-7004&#93;</span> Make /buildWithParameters support remote cause and
user supplied cause text for build via authentication token, just as /build does.</p>
</pre>
</details>

</ul>
</details>

<p>Code changed in hudson<br/>
User: : mindless<br/>
Path:<br/>
 trunk/hudson/main/core/src/main/java/hudson/model/AbstractProject.java<br/>
 trunk/hudson/main/core/src/main/java/hudson/model/ParametersDefinitionProperty.java<br/>
 trunk/hudson/main/core/src/main/resources/hudson/model/BuildAuthorizationToken/config.jelly<br/>
 trunk/www/changelog.html<br/>
<a href="http://jenkins-ci.org/commit/33032" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/33032</a><br/>
Log:<br/>
<span class="error">&#91;FIXED JENKINS-7004&#93;</span> Make /buildWithParameters support remote cause and<br/>
user supplied cause text for build via authentication token, just as /build does.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/amonsen.jpg" title="amonsen's avatar" /> amonsen:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-7004?focusedId=170203&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-170203">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>FYI, it appears <tt>cause</tt> <em>must</em> follow <tt>token</tt> in the query string or cause is discarded. For example: </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">?token=x&amp;cause=y</pre>
</div></div>
<p> works, but </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">?cause=y&amp;token=x</pre>
</div></div>
<p> discards <tt>cause</tt>.</p>
</pre>
</details>

</ul>
</details>

<p>FYI, it appears <tt>cause</tt> <em>must</em> follow <tt>token</tt> in the query string or cause is discarded. For example: </p>

<pre>
?token=x&cause=y</pre>
<p> works, but </p>

<pre>
?cause=y&token=x</pre>
<p> discards <tt>cause</tt>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/solvingj.svg" title="solvingj's avatar" /> solvingj:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-7004?focusedId=359773&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-359773">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We've done some testing and reviewed the commits, and the Cause field does not seem to be fully supported on parameterized builds.   As <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nakilon" class="user-hover" rel="nakilon">nakilon</a> has said, the `cause` field does not work at all when using http auth via header or URL.  The job MUST have a `token` assigned, and as <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=amonsen" class="user-hover" rel="amonsen">amonsen</a> has said, it MUST come before the `cause` parameter.</p>

<p>I suggest reopening this ticket.</p>
</pre>
</details>

</ul>
</details>

<p>We've done some testing and reviewed the commits, and the Cause field does not seem to be fully supported on parameterized builds.   As <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nakilon" class="user-hover" rel="nakilon">nakilon</a> has said, the `cause` field does not work at all when using http auth via header or URL.  The job MUST have a `token` assigned, and as <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=amonsen" class="user-hover" rel="amonsen">amonsen</a> has said, it MUST come before the `cause` parameter.</p>

<p>I suggest reopening this ticket.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>cvijayan:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-7004?focusedId=362613&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-362613">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>My intent is to figure out if there is a way to monitor a job invocation accurately by either using the Queue ID returned as part of the Location and then retrieving the Build ID  OR by passing a unique identifier to the Cause parameter.</p>

<p>What I find weird is that when there is a Quiet period in effect neither the Queue ID nor the Cause parameter behave as expected. Please see the attached sample test harness and its output. </p>

<p>There are 3 jobs invoked in parallel each a unique identifier for cause. This is is the outcome.</p>

<p>A. First job gets a successful build, the other 2 don't (presumably because of the Quiet period).</p>

<p>B.  The First job is queryable through the Queue ID and the response does have the unique identifier passed in Cause. </p>

<p>C. However the second and third invocations get a status of 201 and have the same Queue ID of the first, with the response having all the 3 causes. </p>

<p> </p>

<p>  </p>
</pre>
</details>

</ul>
</details>

<p>My intent is to figure out if there is a way to monitor a job invocation accurately by either using the Queue ID returned as part of the Location and then retrieving the Build ID  OR by passing a unique identifier to the Cause parameter.</p>

<p>What I find weird is that when there is a Quiet period in effect neither the Queue ID nor the Cause parameter behave as expected. Please see the attached sample test harness and its output. </p>

<p>There are 3 jobs invoked in parallel each a unique identifier for cause. This is is the outcome.</p>

<p>A. First job gets a successful build, the other 2 don't (presumably because of the Quiet period).</p>

<p>B.  The First job is queryable through the Queue ID and the response does have the unique identifier passed in Cause. </p>

<p>C. However the second and third invocations get a status of 201 and have the same Queue ID of the first, with the response having all the 3 causes. </p>

<p> </p>

<p>  </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>cvijayan:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-7004?focusedId=362614&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-362614">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Should I open a new ticket or reopen this ticket  ?</p>
</pre>
</details>

</ul>
</details>

<p>Should I open a new ticket or reopen this ticket  ?</p>
